language-support/ts/defs.bzl

```diff
@@ -48,31 +51,19 @@ def da_ts_library(
         for ext in [".js", ".d.ts"]
         for s in srcs
     ]
-    tsc(
-        name = "_%s_tsc" % name,
-        data = [tsconfig] + srcs + deps,
-        outs = outs,
-        args = [
-            "--outDir",
-            "$(RULEDIR)",
-            "--project",
-            "$(location %s)" % tsconfig,
-            "--declaration",
-        ],
-        **kwargs
+    ts_project(
+        name = name + "_tsc",
+        srcs = srcs,
+        tsconfig = tsconfig,
+        deps = deps,
+        tsc = "@language_support_ts_deps//typescript/bin:tsc",
+        validate = False,
+        **kwargs,
     )
-
-    # rules_nodejs does import remapping based on the module_name attribute.
-    # The tsc macro is an instance of npm_package_bin under the covers which
-    # doesn't take a module_name attribute. So, we use this wrapper rule to be
-    # able to set the module_name attribute.
     _da_ts_library_rule(
         name = name,
-        srcs = outs,
-        # We don't do anything with the deps, but they are needed for
-        # rules_nodejs's tracking of transitive dependencies.
-        deps = deps,
```

Comment:
Was the whole `module_name|module_root` feature removed in `rules_nodejs` from 1.1.0 to 1.6.0? Otherwise, if you retain that attribute, would that help?

Reply:
It is still present. I think they want to move everyone over to `LinkablePackageInfo` but for now both should work. I did retain the attributes on `_da_ts_library_rule` (or at least I tried to, definitely possible that I messed it up). I also tried keeping the attributes while removing `LinkablePackageInfo` with no success.